### ISSUE TITLE
**[PROPOSAL]** Escaping

### ISSUE THREAD

**Issue by Erikvv** [spatie/html-element#7]
**In raising this issue, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide]().
- [x] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.
- [x] I have checked that the bug-fix I am reporting can be replicated.

---

### Description of the problem

One of the important reasons for using hyperscript is that it handles escaping of characters which are reserved in html. Which prevents XSS. 

spatie/html-element does not do this. Mainly because it uses strings as intermediate format instead of an object representation of the DOM.

**Comment by sebastiandedeyne** [comment]
I agree. This package isn't being actively developed at the moment, but I'd accept a PR that takes cares of this.

**Comment by holisticnetworking** [comment]
@sebastiandedeyne So is this script effectively dead, development-wise?

**Comment by sebastiandedeyne** [comment]
In terms of active development, yes. We'll still happily tag maintenance or small feature releases contributed by the community.

**Comment by sebastiandedeyne** [comment]
This is partially handled with #11

**Comment by mschop** [comment]
@Erikvv 

I wrote a lib, that does comparable stuff: https://packagist.org/packages/mschop/notee (or the old lib: https://packagist.org/packages/mschop/noteephp).

Compared to spatie/html-element it uses an immutable object representation of the HTML and was developed with a focus on security. It also prevents dangerous URI's like 'src="javascript:malicious()"'.

It also supports template inheritance, events and a block system.

It is currently under development but I will finish version 1.0.0 soon. Emmet support is planned.

**Comment by Erikvv** [comment]
@mschop interesting

In the mean time I also made an implementation https://gitlab.com/erikvv/hyperscript-php but on top of DOMElement

I will look if I can learn anything from your code.
